mdjs-layout/src/mdjs-layout.ts

```diff
@@ -1,91 +1,319 @@
+    } else {
+      this.removeAttribute('data-has-navigation');
+    }
   }
 
-  toggleColorScheme(): void {
+  private renderIsNavigationShown() {
+    if (this._isNavigationShown) {
+      this.setAttribute('data-is-navigation-shown', '');
+    } else {
+      this.removeAttribute('data-is-navigation-shown');
+    }
+    if (this.$navigationToggle) {
+      this.$navigationToggle.setAttribute(
+        'aria-label',
+        `Press to ${this.isNavigationShown ? 'close' : 'open'} navigation`
+      );
+      this.$navigationToggle.innerText = this.isNavigationShown ? '✕' : '☰';
+    }
+  }
+
+  private getLogoHref() {
+    if (this.hasNavigation && this.context.pagesGraph[0]) {
+      return this.relativeUrl(
+        this.context.pagesGraph[0].children
+          ? this.context.pagesGraph[0].children[0].page.url
+          : this.context.pagesGraph[0].page.url
+      );
+    } else {
+      return '';
+    }
+  }
+
+  private get hasNavigation(): boolean {
+    return this.context && this.context.pagesGraph;
+  }
+
+  private relativeUrl(url: string): string {
+    return [
+      Array(url.split('/').length - 1)
+        .fill('..')
+        .join('/'),
+      url,
+    ].join('/');
+  }
+
+  private toggleColorScheme(): void {
     this.colorScheme = this.colorScheme === 'dark' ? 'light' : 'dark';
     this.dispatchColorSchemeChange();
     localStorage.setItem('colorScheme', this.colorScheme);
   }
 
-  toggleNavigation(): void {
+  private toggleNavigation(): void {
     this.isNavigationShown = !this.isNavigationShown;
   }
 
-  @Listen('keydown', { target: 'window' })
-  onWindowKeydown(event: KeyboardEvent): void {
+  private onWindowKeydown(event: KeyboardEvent): void {
     if (this.isNavigationShown && event.key === 'Escape') {
       this.hideNavigation();
     }
   }
 
-  onNavigationWrapperClick(event: Event): void {
+  private onNavigationWrapperClick(event: Event): void {
     if (event.currentTarget === event.target) {
       this.hideNavigation();
     }
   }
 
-  hideNavigation(): void {
+  private hideNavigation(): void {
     this.isNavigationShown = false;
   }
 
-  get hasNavigation(): boolean {
-    return this.context && this.context.pagesGraph;
+  private get isNavigationShown(): boolean {
+    return this._isNavigationShown;
   }
 
-  relativeUrl(url: string): string {
-    return [
-      Array(url.split('/').length - 1)
-        .fill('..')
-        .join('/'),
-      url,
-    ].join('/');
+  private set isNavigationShown(value: boolean) {
+    this._isNavigationShown = value;
+    this.renderIsNavigationShown();
+  }
+
+  private get colorScheme(): ColorScheme {
+    return this._colorScheme;
+  }
+
+  private set colorScheme(value: ColorScheme) {
+    this._colorScheme = value;
+    this.renderColorScheme();
   }
 
-  dispatchColorSchemeChange(): void {
-    this.colorSchemeChange.emit({ colorScheme: this.colorScheme });
+  private renderTemplate() {
+    return /*html*/ `
```

Comment:
I prefer when render is splitted with multiple component function or renderFunction

Reply:
the native version grew way too far in size and indeed it's better to split it
like I said earlier, for private parts of the component I have a rule to split when the file becomes more than 100-150 LOC, which wasn't the case for Lit and StencilJS version of it, but is the case now
I'm doubting though, because the goal to go back to StencilJS version once StencilJS build is ready, otherwise rewriting takes more time which is mostly wasted